test/distributed/test_symmetric_memory.py

```diff
@@ -364,18 +364,21 @@ def test_fused_all_gather_matmul(self, gather_dim: int) -> None:
         A_shard = torch.rand(BATCH, M // self.world_size, K, device="cuda")
         Bs = [torch.rand(K, N, device="cuda") for _ in range(3)]
 
-        ag_output_0, mm_outputs_0 = _fused_all_gather_matmul_fallback(
-            A_shard, Bs, gather_dim=gather_dim, group_name=group.group_name
-        )
-        ag_output_1, mm_outputs_1 = torch.ops.symm_mem.fused_all_gather_matmul(
-            A_shard, Bs, gather_dim=gather_dim, group_name=group.group_name
-        )
+        ag_output_vec = []
+        mm_outputs_vec = []
+        for context in test_contexts:
+            with context():
```

Comment:
nit: it seems a little bit obscure to wrap the test in a loop (and use vector).

Since there is only 1 special context here, how about:
```
    ag_output_0, mm_outputs_0 = torch.ops.symm_mem.fused_all_gather_matmul(
        A_shard, Bs, gather_dim=gather_dim, group_name=group.group_name
    )

    with _test_mode():
        ag_output_1, mm_outputs_1 = torch.ops.symm_mem.fused_all_gather_matmul(
            A_shard, Bs, gather_dim=gather_dim, group_name=group.group_name
        )

    assert torch.allclose(ag_output_0, ag_output_1)
```

Reply:
(Like you did for the other test below)